Review Code

llabake opened the pull request.

llabake committed:
add route to check isbn exist and update user route jsdoc

llabake committed:
add reges check for year, host url const and get user from local storage

llabake committed:
remove unused function and get user from local storage

llabake committed:
remove Wrapper class

llabake committed:
add add book component

llabake committed:
add host url const

llabake committed:
add fetch book action

llabake committed:
add text area input

llabake committed:
add add book component

llabake committed:
add token to header

llabake committed:
change react life cycle method

llabake committed:
Merge pull request #58 from llabake/ft-addBook-react-redux-154038634

Ft add book react redux 154038634

llabake committed:
modify vote response message to include book title

llabake committed:
add caption to revie model and update the resulting test

llabake committed:
include book to add revie response and add get all book revie functionality

llabake committed:
add get all revie on a book endpoint

llabake committed:
update add revie validation on server and client

llabake committed:
add caption to get single book attributes

llabake committed:
add route to get single book component

llabake committed:
add caption to test

llabake committed:
get book after creating revie

llabake committed:
add caption to revie model

llabake committed:
create single book component and the corresponding reducer and actions

llabake committed:
Merge pull request #59 from llabake/ft-get-book-react-redux-154038364

Ft get book react redux 154038364

llabake committed:
install react-materialize and setalert

llabake committed:
add functionality to delete book

llabake committed:
change the response book to camel case

llabake committed:
add all book properties to get single book

llabake committed:
add update root reducer

llabake committed:
create admin component to handle admin operation and its corresponding redux flo

llabake committed:
update action types ith admin actions

llabake committed:
add admin dashboard component to app componenet

llabake committed:
add validation functionality for modify book

llabake committed:
add actions to borro and return book

llabake committed:
remove set authorization function call

llabake committed:
import sign user action by default

llabake committed:
add number proptype to text input value

llabake committed:
add functionality for admin to accept library request

llabake committed:
modify some nav links to render for only admin

llabake committed:
clean action import line

llabake committed:
add functions for update and borro book actions

llabake committed:
solve es lint case literal issue

llabake committed:
add borrocount and fix typo in book controller

llabake committed:
move addbook component and delete authorizatio file

llabake committed:
Merge pull request #60 from llabake/ft-build-admin-component-154179187

Ft build admin component 154179187

llabake committed:
rename borro action to admin action

llabake committed:
rename signup and signin action to user action

llabake committed:
add controller function and route to get users borroed book history

llabake committed:
add user dash board component

llabake committed:
create user component and its corresponding action and reducer

llabake committed:
add user dashboard action types

llabake committed:
add link to profile page

llabake committed:
rename actions

llabake committed:
add link to profile page

llabake committed:
Merge pull request #61 from llabake/ft-userProfile-react-redux-154423282

Ft user profile react redux 154423282

llabake committed:
add empty borroed book to response

llabake committed:
add favorite page component to app

llabake committed:
add favorite action

llabake committed:
add favorite reducer and its initial state

llabake committed:
include book to get user favorite function

llabake committed:
add user favorite page c
omponent

llabake committed:
add unfavorite action and the const

llabake committed:
add remove from favorite operation

llabake committed:
update user reducer to manage unfavorite book  action

llabake committed:
Merge pull request #62 from llabake/ft-favorite-books-page-react-redux-154496475

Ft favorite books page react redux 154496475

llabake committed:
add image field to user model

llabake committed:
add log out action const

llabake committed:
log out action

llabake committed:
clean up user reducer

llabake committed:
add log out functionality

llabake committed:
Merge pull request #63 from llabake/ft-logout-user-154567758

Ft logout user 154567758

llabake committed:
add quantity and include message in response

llabake committed:
add search component and update book reducer and action

llabake committed:
addsearch bar component to header

llabake committed:
remove user from map state to props function

llabake committed:
protect route

llabake committed:
add component to protect route

llabake committed:
protect route

llabake committed:
add search result route

llabake committed:
format home page component

llabake committed:
fix search book response test

llabake committed:
Merge pull request #64 from llabake/ft-search-book-list-154621989

Ft search book list 154621989

llabake committed:
add function to edit review with the corresponding route and controller function

llabake committed:
change anchor tag to link tag

llabake committed:
add initial state for edit revie

llabake committed:
add edit revie component and the corresponding input validation method

llabake committed:
add action and reducer for edit revie component

llabake committed:
add user image

llabake committed:
add functionality to edit and delete revie

llabake committed:
Merge pull request #65 from llabake/ft-delete-and-modify-review-154746870

Ft delete and modify review 154746870

llabake committed:
add operation to upload book image while creating book

llabake committed:
Merge pull request #66 from llabake/ft-upload-image-to-cloud-155002711

add operation to upload book image while creating book

llabake committed:
Serve bundled js files on express server

llabake committed:
Implement popular books endpoint and UI

llabake committed:
Add loading ajax image

llabake committed:
Remove harcoded top user favorites
Modify bookcard

llabake committed:
Add endpoints for popular books and top favorite books

llabake committed:
Fix link to home page
Remove console.log statements
Remove unnecessary toast messages

llabake committed:
add loading status to user favorite page

llabake committed:
add loading to home page and single book page

llabake committed:
add loading status to render

llabake committed:
add action to fetch most popular books and most users favorites books

llabake committed:
add js doc to component did mount

llabake committed:
add loading, top favorite and most popular books initial state

llabake committed:
style post revie button

llabake committed:
style save and cancel button

llabake committed:
add loading status

llabake committed:
add js docs

llabake committed:
add handle cancel function

llabake committed:
add action to handle cancel click in edit revie component

llabake committed:
manange state on cancel action click

llabake committed:
Merge pull request #67 from llabake/ft-improve-app-ui-ux-155112396

Ft improve app ui ux 155112396

llabake committed:
add edit user profile component

llabake committed:
add edit profile to user controller

llabake committed:
add api functionality to get and modify user profile

llabake committed:
add user profile inital state and fetch user operation in the user reducer

llabake committed:
add user profile actions

llabake committed:
add edt profile component

llabake committed:
add imput validation to modify user profile

llabake committed:
add uploaded user image to profile

llabake committed:
add build file to dist folder

llabake committed:
Merge pull request #68 from llabake/ft-edit-user-profile-155312370

Implement editing of user profile - 155312370

llabake committed:
add babel-preset-env as dependency

llabake committed:
add stage-0 to dependecies

llabake committed:
add babel-preset-react to dependecies

llabake committed:
add build logging hosturl and node-env

llabake committed:
use heroku url as host url

llabake committed:
add dist folder

llabake committed:
use https for book default image url

llabake committed:
add dist folder

llabake committed:
modify the rendering of book image

llabake committed:
Merge pull request #69 from llabake/bg-fix-heroku-deploy-155397242

Bg fix heroku deploy 155397242

llabake committed:
add client side test dependencies

llabake committed:
add tests for book reducer

llabake committed:
modify favorite card and favorite card rendering

llabake committed:
handle loading with the approriate message

llabake committed:
remove hardcoded pagination

llabake committed:
implement pagination on booklist

llabake committed:
update book reducer tests

llabake committed:
set state accurated in delete revie reducer

llabake committed:
add js-docs

llabake committed:
add loading to pending borro request

llabake committed:
add pagination to book borro component

llabake committed:
add loading to pending return request

llabake committed:
change response at get all borroed books

llabake committed:
add pagination and remove unused toast message import

llabake committed:
update response data

llabake committed:
add pagination

llabake committed:
add pagination and remove unused link  import

llabake committed:
add pagination and remove unused link  import

llabake committed:
remove empty all books array

llabake committed:
add user reducer tests

llabake committed:
add admin reducer test

llabake committed:
update returned book accept test

Commit:
add set up for actions tests

Commit:
split save book if image exist

Commit:
set up webpack to define global constants

Commit:
export set current user for use in app entry point

Commit:
undo hard coded host url

Commit:
add redux dev tools extension

Commit:
set up local storage mocking

Commit:
format date using moment

Commit:
add state to change colors of icon

Commit:
install moment js

Commit:
add git ignore

Commit:
remove add  from upstream and added it to git ignore file

Commit:
run build

llabake committed:
Merge pull request #70 from llabake/ch-improve-ui-design-155549910

Ch improve ui design 155549910

llabake committed:
Merge pull request #71 from llabake/ch-write-tests-for-reducers-155432919

Ch write tests for reducers 155432919

llabake committed:
install redux logger and global materialize mock

llabake committed:
modify user sign in action
modify user reducer to remove dry code and the modified actions response
remove hard coded user object in the initial state

llabake committed:
chore(test) write actions test
- include user model and favorited book to favorite controllwe
- return borrowed book in the borrow function of borrow controller
- update user object in user data for user reducer test
- change favorite book variable name
- fix user sign up and sign in action and reducer
- add test for actions
- add materialize mock
- get user from the reducer in allBooks component
- clean up sign up with component will recieve props

llabake committed:
Merge pull request #72 from llabake/ch-write-tests-for-actions-155745456

Ch write tests for actions 155745456

9jaswag commented:
💪  great job on what you've done so far. Be sure to check for all instances of the feedback all across your app

llabake committed:
refactor server side code base

llabake committed:
refactor server side code base

Commit:
fix merge conflict

llabake committed:
Merge pull request #74 from llabake/ch-refactor-server-side-156308708

Ch refactor server side 156308708

Commit:
chore(demo feedback): implement first demo feedback
update README file
modify server test start script
modify and remove dry code in controllers
add validations for edit profile, request book, and edit review
add function to paginate result
update paginated url function
use one middleware to find resource and remove dry code
add CONTRIBUTING file
update eslintrc file
[Starts #156841268]

llabake committed:
Create LICENSE

llabake committed:
Merge pull request #75 from llabake/add-license-1

Create LICENSE

llabake committed:
Merge pull request #76 from llabake/ch-implement-demo-feedback-one-156841268

chore(demo feedback): implement first demo feedback

Commit:
chore(test): refactor server side test
seed normal users
update script to run test
add more mock data
use return statement in responses
gitignore DS.store file
update eslintrc globals and rules
add more test cases
add eslint ignore file

[#Starts 157099680]

llabake committed:
Merge pull request #77 from llabake/ch-refactor-server-side-test-157099680

chore(test): refactor server side test

Commit:
chore(webpack): configure webpack per environment
add common webpack file
add development webpack configuration
add production webpack configuration
install webpack plugins

[#Finishes 157289275]

llabake committed:
Merge pull request #78 from llabake/ch-setup-webpack-for-production-157289275

chore(webpack): configure webpack per environment

Commit:
chore(refactor Home Page):
change slider images
modify sign in form page
modify home page
modify slider
update user actions and reducer
update book actions and reducer
join vote model and book model
modify get all books pagination

chore(refactor Home Page):
update actions for get all books and get single book page
modify pagination for get all books page

chore(refactor Home Page):
join vote model and book model
set maximum page limit in utils
update vote and book controller

chore(reactor Home Page):
add count to initial state
modify book reducer to manage set count action
modify user reducers to set current user
update environment variable in webopack

llabake committed:
Merge pull request #79 from llabake/ch-refactor-client-side-codebase-157161392

chore(refactor Client Side Pages): Sign up, All books

llabake committed:
Fix user object on single page

llabake committed:
Fix user object on single page

llabake committed:
Merge pull request #80 from llabake/bg-fix-single-page-user-157392606

Bg fix single page user 157392606

llabake committed:
feature(404 Page):
update route with 404 page component
remove commented code in App.js
remove commented code in search component
create 404 page component

llabake committed:
feature(protect route):
add authentication higher order component
delete protect route component
change font awesome cdn
use authentication HOC in protected routes

[Finishes #157527623]

llabake committed:
Merge pull request #82 from llabake/ft-protect-route-with-HOC-157527623

feature(protect route): using Higher order component

llabake committed:
bug(NavBar-Footer):
fix jumping footer
style to footer component
seperate navbar based on role and authentication
change existing navigation bar in components

[Starts #157534616]

llabake committed:
feature(404 Page):
redirect to 404 for single book not found
update book reducer with resource not found variable

llabake committed:
bug(user object): update user object in component
remove trailing slash from user's borrowed book url
update user object on profile page
update user object on favorite page
fix broken profile page url for user's borrowed books
paginate favorite page in controller

[Finishes #157538479]

llabake committed:
Merge pull request #83 from llabake/bg-fix-footer-navbar-157534616

bug(NavBar-Footer): Fix Nav Bar and Jumping Footer

llabake committed:
--amend

llabake committed:
Merge pull request #84 from llabake/ft-404-page-157527617

Feature(404 page) Implement 404 Page

llabake committed:
Merge pull request #85 from llabake/bg-fix-profile-and-favorite-user-157538479

bug(user object): update user object in component

llabake committed:
(chore): improve application UI/UX
add function to call tool tip
fix responsive table
add cursor pointer to actionable icons and buttons
make book card image link to the book detail page
make carousel image link to the corresponding detail page
make search icon clickable
update review image
fix single book tabs
use materialize truncate class on book overview of book card
update admin reducer with add book operation
update user image on associated user review object
add username to favorite controller user object
update styles
improve profile page side bar
fix search by isbn in search placeholder
fix tabs in admindashboard and single book component
format dates in admindashboard
fix borrowed book url in user action

[finishes #157790385]

llabake committed:
chore(third feedback):
update user active status in user controller
make search case insensitive
use authenticate user function to get user and authenticated initial state
add helper function to verify token
make axios default option a function
add admin dashboard link to admin navigation bar
update sign in and sign up form
update HOC to use authenticate user function
update actions with axios default options function
remove babel polyfill from webpack
add build script to heroku post build script
update webpack common configuration with env config

[finishes #157765415]

llabake committed:
install selenium server
install chrome dev server
add night watch config
write test for home page
write test for sign up page
write test for sign in page
write test for user operations

llabake committed:
Merge pull request #86 from llabake/ch-implement-demo-feedback-three-157765415

chore(third feedback): Implement third demo feedback

llabake committed:
Merge pull request #87 from llabake/ch-improve-app-ui-157790385

(chore): improve application UI/UX

llabake committed:
bug(pagination count): fix pagination set count error
add distinct true to all find and count all query

[finishes #157975743]

llabake committed:
Merge pull request #88 from llabake/bg-fix-pagination-set-count-157975743

bug(pagination count): fix pagination set count error

llabake committed:
(bug): fix edit profile page
redirect to profile page after edit action is successful
disable return button after user attemps to return a book
await reloaded book return request

[finishes #157995723]

llabake committed:
bug(input validation): modify validation
add helper function to check if an input is defined or not empty
remove commented code
add jsdocs
modify jsdocs
validate search input term and reload page if search term is an empty string
remove error outline icon from error div

[starts #158001997]

llabake committed:
chore(update reducer and action test):
update the reducer test file with the latest changes on the application
update action test with the latest changes on the application

[finishes #158007568]

llabake committed:
chore(paginate pages):
implement front end pagination on favorite book page
add front end pagination to user profile page
add front end pagination to admin dashboard page
add action type to hold response count
manage state of response count in their respective reducer
add maximum page limit to webpack common

[starts #157932781]

llabake committed:
fix favorite book pagination delete action

llabake committed:
Merge pull request #89 from llabake/bg-fix-editprofile-page-157995723

(bug): fix edit profile page

llabake committed:
chore(side nav): add side nav component
update eslintrc file env to include jquery
add javascript method to open and collapse side nav
add side nav component to header component
fix jsdocs
fix authenticated status on sign up form

[finishes #158107151]

llabake committed:
Merge pull request #90 from llabake/ch-side-nav-158107151

chore(side nav): add side nav component

llabake committed:
bug(fix side nav class):
change class to classname

[finishes #158109762]

llabake committed:
bug(fix side nav): add side nav proptypes
add react plugin to eslint file

llabake committed:
Merge pull request #91 from llabake/bg-fix-sidenav-class-158109762

bug(fix side nav class):

llabake committed:
bug(input validation):
make errors display one at a time
fix published year server valdiation

llabake committed:
chore(pagination): display new pagination set on delete of an item on favorite page
render new list and page count of reduction of pagination data content
get current state of admin reducer and reduce the count for admindashboard pagination meta
fix proptype validation error on side nav component

[finishes #157932781]

llabake committed:
Merge pull request #92 from llabake/ch-paginate-pages-157932781

Ch paginate pages 157932781

llabake committed:
chore(Loading Component): build loading component
add loading component to all books page
add loading component to favorite page
add loading component to home page component
add loading component to single book component

[finsihes #158150719]

llabake committed:
chore(Loader component): rename loading to loader
add Loader to carousel page

llabake committed:
Merge pull request #93 from llabake/ch-build-loading-component-158150719

chore(Loading Component): build loading component

llabake committed:
chore(Nothing Found Component): build component to be displayed when there is nothing to be displayed
add nothing found component to user favorite page
add nothing found component to admin dashboard page tabs
add nothing found component to all books page

[finishes #158154093]

llabake committed:
Merge pull request #94 from llabake/ch-nothing-found-display-158154093

chore(Nothing Found Component): build component to be displayed when …

llabake committed:
bug(mixed content): change image url
add two more books to book seed data
change image url from http to https

[finishes #158165879]

llabake committed:
Merge pull request #95 from llabake/bg-mixed-content-158165879

bug(mixed content): change image url

llabake committed:
Merge pull request #96 from llabake/bg-fix-input-validation-158001997

Bg fix input validation 158001997

llabake committed:
bug(stabilize app): style search bar close icon
add validation for first name and last name length

[starts #158178673]

llabake committed:
bug(stabilize app):sort reviews in descending order
add bookId to edit review and delete review route

llabake committed:
bug(stabilize app): style active tab

llabake committed:
bug(stabilize app): put nothing found component in a container
add bookId in method to modify book
format review created at dated
add bookId in method to delete book

[finishes #158178673]

llabake committed:
Merge pull request #97 from llabake/bg-stabilize-application-158178673

Bg stabilize application 158178673

llabake committed:
feature(request book page): add function to get suggested books
seed request book data
add route to get requested books

[starts #158186035]

llabake committed:
(request book page): add request book component
build table row not found component
add action to fetch requested books
manage state of request book by updating admin reducer with request book actions
add suggestedBooks and suggestedBookCount to the initial state
add test for requested book page action and reducer

[finishes #158186035]

llabake committed:
Merge pull request #98 from llabake/ft-request-book-page-158186035

Ft request book page 158186035

llabake committed:
chore(components): change components to JSX files

[finishes #158239827]

llabake committed:
Merge pull request #99 from llabake/ch-change-components-to-jsx-158239827

chore(components): change components to JSX files

llabake committed:
chore(fourth feedback):remove console log statements
fix saerch bar unique id error
make tabs clickable
re arrange single book page and order review sectio before carousel
remove unused images

[finshes #158240163]

llabake committed:
Merge pull request #100 from llabake/ch-fourth-feedback-158240163

chore(fourth feedback):implement fourth feedback

llabake committed:
chore(test component): write test for common components

[starts #158038920]

llabake committed:
Merge pull request #101 from llabake/ch-update-action-and-reducers-test-158007568

chore(update reducer and action test):

llabake committed:
chore(api documentation): update API documentation
update Readme file with front end test script

[finishes #158409247]

llabake committed:
Merge pull request #103 from llabake/ch-api-documentation-158409247

chore(api documentation): update API documentation

llabake committed:
(E2E test): write e2e test for all user interation
add test for admin operation

finishes [#157898365]

llabake committed:
Merge pull request #104 from llabake/ch-end-to-end-test-157898365

Ch end to end test 157898365

llabake committed:
Merge pull request #102 from llabake/ch-write-component-test-158038920

chore(test component): write test for common components

llabake committed:
(Front End Test): Fix failing admin action test
increase wait time for view all review assertion
fix nav bar dropdown li

[finishes #158472973]

llabake committed:
Merge pull request #105 from llabake/bg-fix-failing-FE-tests-158472973

(Front End Test): Fix failing front end test
